### DISCUSSION TITLE
Will there be a canonical unused-renders mod on the Factorio mod portal?

### DISCUSSION THREAD

**Discussion by DoubleThoughtTheProgrammer** [malcolmriley/unused-renders#7] [General]
In order to prevent duplication of textures, is there (or will there be) a canonical unused-renders mod? Here's a pros/cons analysis of doing so:

Pros:
- It will help prevent asset duplication. It can be bad for VRAM if the same high-resolution textures are loaded multiple times from different files. By putting the files in a canonical graphics mod, all references to the texture files in the canonical mod will reuse the same textures, saving on VRAM.
- It will ease the legal burden (from section 3 of the license) on mod developers using these graphics from the canonical mod, because you can declare that using the canonical mod as a prerequisite satisfies all the requirements of Section 3(a). Without a canonical mod, it is easy to accidentally be in noncompliance with the license (by forgetting a part of the required attribution).

Cons:
- Mod authors who use the canonical mod without using all the graphics waste their users' bandwidth (when downloading the mod) and storage (when storing the mod)
- You'll have to eventually update the mod when you add new graphics to the repository

**Comment by malcolmriley** [comment]
Hello,

Thank you for your message and the thought you put into this. I have considered your points and my answer to the question at the outset is _"Not for the foreseeable future."_ It is my opinion that the benefits of a standalone graphics mod are (at this time) quite small in comparison with the drawbacks you mentioned.

By what I admit to be an estimation, the asset duplication issue will only be problematic if the user is using multiple mods that each provide an unedited version of the same image. Even then, unless the set of mods providing copies of the same images share _many_ such images, I can't imagine the impact on VRAM of this duplication is going to be noteworthy. Most mod authors seem to prefer "bespoke" art for their projects; I don't foresee a future where this repository becomes so overused as to become a problem for mod-user VRAM.

I consider the concerns surrounding legal burden to be a nonissue for both a practical and a selfish reason:
- **The Practical**: The images provided by this repository have authorship information embedded into the metadata. Therefore, I consider anyone using an unedited version of an image from this repository to be minimally in compliance automatically. Only a hypothetical user that modifies the image in such a way to remove this metadata (either accidentally or purposefully) and then neglects to provide external attribution would be in violation of the license - and in that case, I refer to the second point:
- **The Selfish**: I am fundamentally uninterested in policing how users adapt my work, even if that means that I won't receive credit in some circumstances. My motivation for choosing an attribution license is not for credit, clout or accolades, but so that users that have benefitted from use of the free resource can pay it forward by making other *prospective* users aware that the free resource exists and is available to them too. The great dream of the internet was of a post-scarcity virtual space where digital resources could be freely exchanged. I am not sure that this dream has been fully realized (and indeed there seems to be some active interest in working against it - particularly the "post-scarcity" and "free" parts), but at the very least I can pay tribute to the idea of it with my own works.

One final note: it doesn't have to be me that produces the shared-assets mod. The license permits that kind of adaptation too, so even if you can't convince me of the necessity of such a project, if you can convince anybody else that it's a good idea, they can freely clone this repository into it, and even monetize it as they see fit!

Thank you again for reaching out - I am happy to continue this discussion and am not immune to having my mind changed, so if you have any further thoughts to contribute, please do so.

**Reply by DoubleThoughtTheProgrammer** [reply]
Thanks for your comment and the insight into your thought process. I admit I'm much worse of a digital artist than I am a programmer, so having a large repository of Factorio-style icons will be extremely useful as I'm working on my mod. I'm very grateful that you've released this repository, and I'm planning to use the heck out of these icons. If you want early access to the mod in development, feel free to shoot me a message. Unfortunately, it's not in a particularly playable state yet, and it's definitely not going to be fully playable until after the 2.0 release.

**Reply by malcolmriley** [reply]
In case you missed it, I've started posting some Blender tutorials [on the wiki section of this repository](https://github.com/malcolmriley/unused-renders/wiki). Producing the most commonly-used imagery (such as plates, beams, ingots, gears, etc) is made very straightforward by Blender, so if you can spare the time I strongly believe it will pay dividends. I also consider myself to be a programmer first and an artist second, so if I can do it, so can you - you will probably surprise yourself with what you can accomplish!

I admit that the tutorial series is a bit lacking at the moment but it should cover most of the basic techniques. I want to return to work on the series but I've been a bit too busy as of late. Regardless, I don't intend to keep any secrets so if you (or any other eavesdropper) have questions about how I made a particular render, feel free to send another message.

Also, please send me a link to your mod as soon as it is in a state where you feel comfortable sharing! I love mods and I enjoy seeing these sprites in use. Good luck with your project!
